docs(en): add docs for including RSS post content (#2129)
withastro/docs: src/pages/en/guides/rss.md

});
```

### Including full post content

By default, the Astro RSS integration does not support including the content of each of your posts in the feed itself. 

> However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key. We suggest using a package like [sanitize-html](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped and encoded for use in the XML feed.

Reviewer: ```suggestion
However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key. We suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded for use in the XML feed.
```

> However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key. We suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded for use in the XML feed.

Reviewer: ```suggestion
However, if you create the list of RSS feed objects yourself, you can pass the `compiledContent` property of Markdown files (not MDX), to the `content` key. We suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded for use in the XML feed.
```

> However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key using the [`compiledContent` property](/en/guides/markdown-content/#exported-properties). We suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded for use in the XML feed.

Reviewer 2: Totally a nit but @sarah11918 should we be using `()` here to help indicate that’s a function you call rather than a static property?

```suggestion
However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key using the [`compiledContent()` property](/en/guides/markdown-content/#exported-properties). We suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded for use in the XML feed.
```

Otherwise, looks great! Thanks for this contribution @smithbm2316 🙌

> ```js ins={2, 15} title={src/pages/rss.xml.js}

Reviewer: ```suggestion
```js ins={2, 16} title={src/pages/rss.xml.js}
```
I added a code comment, so I believe this number has to increase by one

});
```

> ### Adding the post's content to the feed
> 
> By default, the Astro RSS integration does not support including the content of each of your posts in the feed itself. However, if you choose option 2 above to create a list of RSS feed objects yourself, you can pass the content of your post to the `content` key to add it to the feed, so long as you pass it HTML that has been properly sanitized and escaped. The `compiledContent` function that is returned on each post from the call to `import.meta.glob` will return the content of your markdown already processed as HTML by Astro. All you need to do at that point is run it through a package like [sanitize-html](https://www.npmjs.com/package/sanitize-html) in order to make sure that it is properly escaped and encoded to use in the XML feed.
> 
> :::note
> This feature is currently only supported with Markdown files, _not_ MDX. See the API for [Astro.glob() (which uses import.meta.glob under the hood](/en/guides/markdown-content/#exported-properties) to see the full list of properties are supported for Markdown, MDX, or both.
> :::
> 
> ```js ins={2, 15} title={src/pages/rss.xml.js}
> import rss from '@astrojs/rss';
> import sanitizeHtml from 'sanitize-html';
> 
> const postImportResult = import.meta.glob('../posts/**/*.md', { eager: true });
> const posts = Object.values(postImportResult);
> 
> export const get = () => rss({
>   title: 'Buzz’s Blog',
>   description: 'A humble Astronaut’s guide to the stars',
>   site: import.meta.env.SITE,
>   items: posts.map((post) => ({
>     link: post.url,
>     title: post.frontmatter.title,
>     pubDate: post.frontmatter.pubDate,
>     content: sanitizeHtml(post.compiledContent()),
>   }))
> });
> ```

Reviewer: ``````suggestion
### Including full post content

By default, the Astro RSS integration does not support including the content of each of your posts in the feed itself. 

However, if you create the list of RSS feed objects yourself, you can pass the content of Markdown files (not MDX), to the `content` key. We suggest using a package like [sanitize-html](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped and encoded for use in the XML feed.

```js ins={2, 15} title={src/pages/rss.xml.js}
import rss from '@astrojs/rss';
import sanitizeHtml from 'sanitize-html';

// Works with Markdown files only!
const postImportResult = import.meta.glob('../posts/**/*.md', { eager: true }); 
const posts = Object.values(postImportResult);

export const get = () => rss({
  title: 'Buzz’s Blog',
  description: 'A humble Astronaut’s guide to the stars',
  site: import.meta.env.SITE,
  items: posts.map((post) => ({
    link: post.url,
    title: post.frontmatter.title,
    pubDate: post.frontmatter.pubDate,
    content: sanitizeHtml(post.compiledContent()),
  }))
});
```
``````

It is so exciting to have a solution in docs for this, @smithbm2316!  🥳 

What do you think about streamlining the text something like above so that the required information for the user stands out a little more?

Author: Brevity is not my strength, I love your revision! So much more concise and covers everything needed that I think needs to be conveyed about this feature! :smile:

Reviewer 3: @smithbm2316 Speaking as a docs review watcher, you're good to commit!

Reviewer: B Holmes Dev in the house! 👀 

Yes, @smithbm2316, it's fine to commit if you're happy with it!

Now that I know it's OK, I might give it another bit of polish. But you can simply commit, and I can take it from here. Thanks again for such a great contribution!  🚀
